sdk/resourcemanager/Azure.ResourceManager/src/ArmClient.cs

```diff
@@ -157,40 +157,50 @@ public virtual TenantContainer GetTenants()
         /// Gets a resource group operations object.
         /// </summary>
         /// <param name="id"> The id of the resourcegroup. </param>
-        /// <returns> Resource operations of the resource. </returns>
-        public virtual ResourceGroupOperations GetResourceGroupOperations(string id)
+        /// <returns> Resource operations of the resourcegroup. </returns>
+        public virtual ResourceGroup GetResourceGroup(string id)
         {
-            return new ResourceGroupOperations(new ClientContext(ClientOptions, Credential, BaseUri, Pipeline), id);
+            return new ResourceGroup(new ClientContext(ClientOptions, Credential, BaseUri, Pipeline), id);
         }
 
         /// <summary>
         /// Gets a subscription operations object.
         /// </summary>
         /// <param name="id"> The id of the subscription. </param>
         /// <returns> Resource operations of the subscription. </returns>
-        public virtual SubscriptionOperations GetSubscriptionOperations(string id)
+        public virtual Subscription GetSubscription(string id)
```

Comment:
These first class Get calls all taking string id as parameter. The expectation may not be clear on the format. For example,  for RG, you would expected the resource id as "sub/id/rg/id" points to the RG. But for subscription, providers etc, just the {id} rather than "/subscription/id" should just work. I suggest we do some string value check and support both cases.

Reply:
maybe change the type to ResourceIdentifier to make it clear we expect a full id in all cases?

Reply:
I would suggest for all non first level resources, the parameter to be ResourceIdentifier. For top-level resources, we can keep it as a string and be more lenient, so it is more convenient to use.